bot/exts/backend/branding/_cog.py

```diff
@@ -104,19 +104,26 @@ class Branding(commands.Cog):
     """
 
     # RedisCache[
-    #     "daemon_active": bool            | If True, daemon starts on start-up. Controlled via commands.
-    #     "event_path": str                | Current event's path in the branding repo.
-    #     "event_description": str         | Current event's Markdown description.
-    #     "event_duration": str            | Current event's human-readable date range.
-    #     "banner_hash": str               | SHA of the currently applied banner.
-    #     "icons_hash": str                | Compound SHA of all icons in current rotation.
-    #     "last_rotation_timestamp": float | POSIX UTC timestamp.
+    #     "daemon_active": bool                   | If True, daemon starts on start-up. Controlled via commands.
+    #     "event_path": str                       | Current event's path in the branding repo.
+    #     "event_description": str                | Current event's Markdown description.
+    #     "event_duration": str                   | Current event's human-readable date range.
+    #     "banners_hash": str                     | Compound SHA of all banners in the current rotation.
+    #     "icons_hash": str                       | Compound SHA of all icons in current rotation.
+    #     "last_icon_rotation_timestamp": float   | POSIX UTC timestamp.
+    #     "last_banner_rotation_timestamp": float | POSIX UTC timestamp.
     # ]
     cache_information = RedisCache()
 
-    # Icons in current rotation. Keys (str) are download URLs, values (int) track the amount of times each
-    # icon has been used in the current rotation.
-    cache_icons = RedisCache()
+    # Icons and banners in current rotation.
+    # Keys (str) are download URLs, values (int) track the amount of times each
+    # asset has been used in the current rotation.
+    _cache_icons = RedisCache()
+    _cache_banners = RedisCache()
+    asset_caches = {
+        AssetType.ICON: _cache_icons,
+        AssetType.BANNER: _cache_banners
+    }
```

Comment:
Why not put the caches directly in the dictionary? I am pretty sure you can manually provide a namespace string.

Reply:
Also, do you think it's worth it to use `MappingProxyType` to make this immutable and prevent programming errors?

Reply:
You can provide it, but it would require hardcoding the class name to get the same behavior. In retrospect, that probably would've been better. I'll change it.

I'll also look into the mapping proxy.

Reply:
Pushed the MappingProxy and Explicit namespaces in 90c34292 and a383040c respectively.

Reply:
Looks good.